core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/internal/marshall/JsonProtocolMarshaller.java

```diff
@@ -177,6 +178,12 @@ void doMarshall(SdkPojo pojo) {
             Object val = field.getValueOrDefault(pojo);
             if (isBinary(field, val)) {
                 request.contentStreamProvider(((SdkBytes) val)::asInputStream);
+            } else if (isDocumentType(field) && val != null) {
+                if (hasExplicitPayloadMember && field.containsTrait(PayloadTrait.class)) {
+                    jsonGenerator.writeStartObject();
```

Comment:
Why is `jsonGenerator.writeStartObject()` called for explicit payload member?

Reply:
In case of implicit payload writeStartObject will be done at
[StartMarshilling](https://github.com/aws/aws-sdk-java-v2/blob/a1f75b3807e548f3fab4300cf450524204a3e374/core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/internal/marshall/JsonProtocolMarshaller.java#L171).

This check gets skipped because we do writeStartObject when the actual payload member is iterated [here]
(https://github.com/aws/aws-sdk-java-v2/blob/a1f75b3807e548f3fab4300cf450524204a3e374/core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/internal/marshall/JsonProtocolMarshaller.java#L182)

writeStartObject on jsonGenerator is required to mark the context as Object which make the wtiteContext ready to accept Object else it always expects a Key Pair.

In DocumentType the members must always will be in payload location (as per specs ) and since writeObject gets skipped we need to perform explicit write object before writing the Object to jsonGenerator.

Reply:
Thanks for offline review.
After the review I had done following modifications

1. WriteEndObject for DocumentType
2. Handle Null response
3. Handle Explicit Responses so that they are directly Unmrashalled as respective document rather than Maps.

Diff is as below

https://github.com/aws/aws-sdk-java-v2/compare/e95ab8ecd488290e31d42a2c9cd491887c9c40ef..2b94c81253a8d09854e5b7261a29c28d6af8e98a